src/components/StylesFromLiteralsService.js

```diff
@@ -219,47 +228,69 @@ goog.provide('ga_stylesfromliterals_service');
         return this.defaultStyle;
       };
 
+      OlStyleForPropertyValue.prototype.setOlText_ = function(olStyle,
+        labelProperty, labelTemplate, properties) {
+        var text;
+        properties = properties || [];
+        if (labelProperty) {
+          text = properties[labelProperty];
+          if (text !== undefined && text !== null) {
+            text = text.toString();
+          }
+        } else if (labelTemplate) {
+          text = labelTemplate;
+          angular.forEach(properties, function(prop, k) {
+            if (prop !== undefined && prop !== null) {
+              text = text.replace('${' + k + '}', prop.toString());
+            }
+          });
+        }
+        if (text) {
+          olStyle.getText().setText(text);
+        }
+        return olStyle;
+      };
+
+      OlStyleForPropertyValue.prototype.getOlStyle_ = function(feature,
+          resolution, properties) {
+        var value, geomType, olStyles, styleSpec;
+        // A value can be 0
+        value = properties[this.key];
+        value = value !== undefined ? value : this.defaultVal;
+        geomType = getGeomTypeFromGeometry(feature.getGeometry());
+
+        if (this.type === 'unique') {
+          olStyles = this.styles[geomType][value];
+        } else if (this.type === 'range') {
+          olStyles = this.findOlStyleInRange_(value, geomType);
+        }
+
+        if (!olStyles) {
+          return this.alertDebug_(value, feature.getId());
+        }
+        styleSpec = this.getOlStyleForResolution_(olStyles, resolution);
+        if (styleSpec) {
+          return this.setOlText_(styleSpec.olStyle, styleSpec.labelProperty,
+              styleSpec.labelTemplate, properties);
+        }
+        return this.defaultStyle;
+      };
+
       OlStyleForPropertyValue.prototype.getFeatureStyle = function(feature,
           resolution) {
-        var properties, value, geomType, olStyles, res, labelProperty, text;
+        var properties;
+        if (feature) {
```

Comment:
In which case a fetaure can be null?

Reply:
In that case https://github.com/geoadmin/mf-geoadmin3/blob/99b4743709268a7619d44289ebec4c3fdd562ac1/test/specs/StylesFromLiterals.spec.js#L28

But you're right if in a style function feature should always be defined. I prefer it to be robust though.

Reply:
Ok it's not so important.